src/controllers/sharedExecutionPlanUtils.ts

```diff
@@ -0,0 +1,184 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { exists } from "../utils/utils";
+import { homedir } from "os";
+import {
+    ExecutionPlanGraphInfo,
+    ExecutionPlanReducers,
+    ExecutionPlanWebviewState,
+} from "../reactviews/pages/ExecutionPlan/executionPlanInterfaces";
+import { ExecutionPlanService } from "../services/executionPlanService";
+import { QueryResultWebviewState } from "../sharedInterfaces/queryResult";
+import * as vscode from "vscode";
+import UntitledSqlDocumentService from "./untitledSqlDocumentService";
+import { ApiStatus } from "../sharedInterfaces/webview";
+import {
+    TelemetryActions,
+    TelemetryViews,
+} from "../sharedInterfaces/telemetry";
+import { sendActionEvent, sendErrorEvent } from "../telemetry/telemetry";
+
+export async function saveExecutionPlan(
+    state: QueryResultWebviewState | ExecutionPlanWebviewState,
+    payload: ExecutionPlanReducers["saveExecutionPlan"],
+) {
+    let folder = vscode.Uri.file(homedir());
+    let filename: vscode.Uri;
+
+    // make the default filename of the plan to be saved-
+    // start at plan.sqlplan, then plan1.sqlplan, ...
+    let counter = 1;
+    if (await exists(`plan.sqlplan`, folder)) {
+        while (await exists(`plan${counter}.sqlplan`, folder)) {
+            counter += 1;
+        }
+        filename = vscode.Uri.joinPath(folder, `plan${counter}.sqlplan`);
+    } else {
+        filename = vscode.Uri.joinPath(folder, "plan.sqlplan");
+    }

```

Comment:
Can you pull this into a helper function in `utils.ts`?  I think it'll be useful elsewhere:

`function getUniqueFilePath(folder, baseName, extension) { ... }`